usaspending_api/agency/v2/views/subcomponents.py

```diff
@@ -110,11 +112,18 @@ def get_file_b_queryset(self):
         return results
 
     def get_common_query_objects(self, treasury_account_keyword):
-
+        latest = (
+            SubmissionAttributes.objects.filter(
+                submission_window__submission_reveal_date__lte=now(), reporting_fiscal_year=self.fiscal_year
+            )
+            .values("reporting_fiscal_year")
+            .annotate(max_fiscal_period=Max(F("reporting_fiscal_period")))
+            .values("max_fiscal_period")
+        )
         filters = [
             Q(**{f"{treasury_account_keyword}__federal_account__parent_toptier_agency": self.toptier_agency}),
             Q(submission__reporting_fiscal_year=self.fiscal_year),
-            Q(submission__reporting_fiscal_period=self.fiscal_period),
+            Q(submission__reporting_fiscal_period=latest[0]["max_fiscal_period"]),
```

Comment:
Should the `fiscal_period` property in AgencyBase be changed instead? If this issue is coming up with the switch to File A/B then won't this potentially affect anywhere using that property?

Reply:
Yeah, theoretically it would, but I was under the impression that this was intentionally based on the comment under the `fiscal_period()` property in agency_base.py:
```
        """
        This is the fiscal period we want to limit our queries to when querying CPE values for
        self.fiscal_year.  If it's prior to Q1 submission window close date, we will return
        quarter 1 anyhow and just show what we have (which will likely be incomplete).
        """
```
Now I don't know if that's still what we want to do, given the situation, but it does seem like we did it on purpose. I also don't want to randomly switch every use of `fiscal_period` on the agency page to a new value without more testing.

Reply:
Screenshot of what I am seeing with your branch against QAT DB
![Screen Shot 2022-02-02 at 10 30 46 AM](https://user-images.githubusercontent.com/40359517/152215718-bcaea6c5-c87f-4db5-bf45-bb6da6cbaa4c.png)

Reply:
I'll update the bureau federal account endpoint, since that was also changed to file a/b, but the rest of the endpoints on the agency page I believe were intended to operate as-is